File: PerfTest2/PerfTest2-Xamarin/PerfTest2-Xamarin.Shared/Utilities/FileUtilities.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace PerfTest2Xamarin.Utilities
{
    public class FileUtilities : IDisposable
    {
        private string filePath;
        private StreamWriter streamWriter = null;

        public FileUtilities(string filePath)
        {
            this.filePath = Path.Combine(filePath, "testFile.txt");
        }

        public void OpenFile() 
        {
            streamWriter = new StreamWriter(filePath);
        }

        public void CloseFile()
        {
            if (streamWriter != null)
            {
                streamWriter.Close();
                streamWriter.Dispose();
                streamWriter = null;
            }
        }

        public void DeleteFile()
        {
            if (File.Exists(filePath))
            {
                File.Delete(filePath);
            }
        }

        public void CreateFile() 
        {
            if (!File.Exists(filePath))
            {
                using (var stream = File.Create(filePath))
                {
                }
            }
        }

        public void WriteLineToFile(String line) 
        {
            if (!File.Exists(filePath)) 
            {
                this.CreateFile();
            }
            if (streamWriter == null) 
            {
                this.OpenFile();
            }

            streamWriter.WriteLine(line);
        }

        public IList<string> ReadFileContents()
        {
            if (!File.Exists(filePath))
            {
                this.CreateFile();
            }

            var returnValue = new List<String>();

            using (var streamReader = new StreamReader(filePath))
            {
                string line;
                while ((line = streamReader.ReadLine()) != null)
                {
                    returnValue.Add(line);
                }
            }
            return returnValue;
        }

        public void Dispose()
        {
            this.CloseFile();
        }
    }
}

File: PerfTest2/PerfTest2-Java/app/src/main/java/com/vandammeford/kevinf/perftest2_java/FileUtilities.java
package com.vandammeford.kevinf.perftest2_java;

import android.content.Context;
import android.os.Environment;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.util.ArrayList;

/**
 * Created by KevinF on 1/16/2015.
 */
public class FileUtilities {

    private String filePath;
    BufferedWriter fileHandle;
    Context context;
    File textFile;

    public FileUtilities(Context context) {
        this.context = context;
        String filePath = context.getExternalFilesDir(Environment.DIRECTORY_DOWNLOADS) + File.separator + "testFile.txt";
        textFile = new File(filePath);
    }

    public void openFile() throws Exception {
        fileHandle = new BufferedWriter(new FileWriter(textFile));
    }

    public void closeFile() throws Exception {
        if (fileHandle != null) {
            fileHandle.close();
            fileHandle = null;
        }
    }

    public void deleteFile() {
        if (textFile.exists()) {
            textFile.delete();
        }
    }

    public void createFile() throws Exception {
        if (!textFile.exists()) {
            textFile.createNewFile();
        }
    }

    public void writeLineToFile(String line) throws Exception {
        if (!textFile.exists()) {
            this.createFile();
        }
        if (fileHandle == null) {
            this.openFile();
        }

        fileHandle.write(line);
    }

    public ArrayList<String> readFileContents() throws Exception {

        BufferedReader reader = new BufferedReader(new FileReader(textFile));
        String line;
        ArrayList<String> returnValue = new ArrayList<String>();

        while((line = reader.readLine()) != null){
            returnValue.add(line);
        }
        reader.close();
        return returnValue;
    }
}